test/Microsoft.AspNet.Mvc.Core.Test/RemoteAttributeTest.cs

```diff
@@ -0,0 +1,678 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.AspNet.Builder;
+using Microsoft.AspNet.Http.Core;
+using Microsoft.AspNet.Mvc.Internal;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Microsoft.AspNet.Routing;
+using Microsoft.Framework.DependencyInjection;
+using Microsoft.Framework.Logging;
+using Microsoft.Framework.OptionsModel;
+using Moq;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc
+{
+    public class RemoteAttributeTest
+    {
+        // Action, controller, and route names containing just whitespace are not legal.
+        public static TheoryData<string> InvalidNames
+        {
+            get
+            {
+                return new TheoryData<string>
+                {
+                    null,
+                    string.Empty,
+                    "   ",
+                    "\t \t",
+                    " \r\n"
+                };
+            }
+        }
+
+        // Property names containing just whitespace are legal.
+        public static TheoryData<string> NullOrEmptyNames
+        {
+            get
+            {
+                return new TheoryData<string>
+                {
+                    null,
+                    string.Empty,
+                };
+            }
+        }
+
+        [Fact]
+        public void IsValidAlwaysReturnsTrue()
+        {
+            // Act & Assert
+            Assert.True(new RemoteAttribute("RouteName", "ParameterName").IsValid(null));
+            Assert.True(new RemoteAttribute("ActionName", "ControllerName", "ParameterName").IsValid(null));
+        }
+
+        [Theory]
+        [MemberData(nameof(InvalidNames))]
+        public void Constructor_WithInvalidRouteName_Throws(string routeName)
+        {
+            // Arrange & Act & Assert
+            Assert.Throws<ArgumentException>("routeName", () => new RemoteAttribute(routeName));
```

Comment:
I'm not sure the specifics of the message are all that interesting in any of the new `_Throws` tests.  but :ok: to checking it.  is that a general request?

:-1: to using the test helper.  looks like we've been directly doing the `var exception = Assert.Throws(...); Assert.Equal(expectedMessage, exception.Message);` in the test methods.  and that's more clear.